pkg/cloud/services/network/vpc.go

```diff
@@ -276,17 +276,8 @@ func (s *Service) deleteVPCEndpoints() error {
 		return nil
 	}
 
-	// Gather all services that might have been enabled.
-	services := sets.New[string]()
-	if s.scope.Bucket() != nil {
-		services.Insert(fmt.Sprintf("com.amazonaws.%s.s3", s.scope.Region()))
-	}
-	if services.Len() == 0 {
-		return nil
-	}
-
 	// Get all existing endpoints.
-	endpoints, err := s.describeVPCEndpoints()
+	endpoints, err := s.describeVPCEndpoints(filter.EC2.Cluster(s.scope.Name()))
```

Comment:
I would say that we want to make sure its `owned` and not `shared`, mainly to be consistent with other places in the code.

```suggestion
	endpoints, err := s.describeVPCEndpoints(filter.EC2.ClusterOwned(s.scope.Name()))
```